### PR TITLE
Exclude accounts which are outside of the retention period

### PR DESCRIPTION
**Pull request by jacobwinch**
... so that we can allow the associated identity accounts to be deleted.

Note that this PR also removes a distinction between users who don't appear in Zuora at all, and those who are in Zuora (but are outside of the relevant retention period), because nobody can think of a use case for this extra detail in the response.

### CODE REVIEW

File: handlers/identity-retention/src/main/scala/com/gu/identityRetention/IdentityRetentionResponses.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@ object IdentityRetentionResponseModels {
     implicit val successResponseWrites = Json.writes[SuccessResponse]
   }
 
-  case class NotFoundResponse(previousRelationship: Boolean) extends IdentityRetentionResponse
+  case class NotFoundResponse(message: String = "identity account can be deleted") extends IdentityRetentionResponse
```
Comment (paulbrown1982) [inline handlers/identity-retention/src/main/scala/com/gu/identityRetention/IdentityRetentionResponses.scala]: I don't agree with this message text - it's not up to us to advise Identity to delete, but to advise that we have no active relationship for the customer it sent to us.

File: handlers/identity-retention/src/main/scala/com/gu/identityRetention/HasActiveZuoraAccounts.scala [hunk 2]
```diff
@@ -11,13 +11,25 @@ import scalaz.{-\/, \/-}
...
-  case class IdentityQueryResponse(Id: String, Status: String)
+  case class IdentityQueryResponse(Id: String)
   implicit val reads = Json.reads[IdentityQueryResponse]
 
   def apply(identityId: IdentityId, zuoraQuerier: ZuoraQuerier): FailableOp[List[AccountId]] = {
 
     def searchForAccounts = {
-      val identityQuery = ZuoraQuery.Query(s"select id, status from account where IdentityId__c = '${identityId.value}'")
+
+      val commonConditions = s"IdentityId__c = '${identityId.value}' and status != 'Canceled'"
```
[the remark this answers is not in the document]
Comment (johnduffell) [reply handlers/identity-retention/src/main/scala/com/gu/identityRetention/HasActiveZuoraAccounts.scala]: how about `nonCancelledForIdentityId` ?

File: handlers/identity-retention/src/main/scala/com/gu/identityRetention/HasActiveZuoraAccounts.scala [hunk 3]
```diff
@@ -30,7 +42,7 @@ object HasActiveZuoraAccounts {
       case \/-(result) if result.size > 0 =>
         \/-(result.records.map(account => AccountId(account.Id)))
       case \/-(result) if result.size == 0 =>
-        -\/(IdentityRetentionApiResponses.notFoundInZuora)
+        -\/(IdentityRetentionApiResponses.canBeDeleted)
```
Comment (johnduffell) [inline handlers/identity-retention/src/main/scala/com/gu/identityRetention/HasActiveZuoraAccounts.scala]: shoudl have changed `canBeDeleted` to be `noActiveRelationships` to match

**Comment by johnduffell** [comment]
yep looks good, just a couple of naming things 👍

**Pull request merged**